geomesa-feature/src/test/scala/org/locationtech/geomesa/feature/kryo/KryoFeatureSerializerTest.scala

```diff
@@ -107,7 +107,44 @@ class KryoFeatureSerializerTest extends Specification {
         val in = new ByteArrayInputStream(out.toByteArray)
         val deserialized = serializer.read(in)
 
-        deserialized must not beNull;
+        deserialized must not(beNull)
+        deserialized.getType mustEqual sf.getType
+        deserialized.getAttributes mustEqual sf.getAttributes
+      }
+    }
+
+    // NB: this doesn't actually seem to cause the error I was seeing, but
+    // ScaldingDelimitedIngestJobTest in geomesa-tools does cause it...
+    "correctly serialize with a type having the same field names but different field types" in {
+      val spec = "a:Integer,m:List[String],l:Map[Double,Date],dtg:Date,*geom:Point:srid=4326"
```

Comment:
[picky]  Is there a reason the list attribute is called ```m``` while the map attribute is called ```l```?  Mind blown.